.github/workflows/integration.yml

```diff
@@ -0,0 +1,106 @@
+name: integration tests
+
+on:
+  workflow_dispatch:
+  workflow_call:
+  pull_request:
+    branches:
+      - develop
+      - releases/**
+
+concurrency:
+  group: intergrations-${{github.ref}}-${{github.event.pull_request.number || github.run_number}}
+  cancel-in-progress: true
+
+jobs:
+  integration-nix:
```

Comment:
nitpick: `nix` is ambiguous and rather odd in the context of Spack the package manager.  I would recommend to avoid that term. Especially since you're only testing Linux and macOS there's no point in talking about "*nix" if that was the intention.